src/slash/deck.ts

```diff
@@ -80,9 +81,134 @@ export class DeckCommand extends AutocompletableCommand {
 			return;
 		}
 
-		const row = generateDeckValidateButtons();
-		const response = await interaction.reply({ content: outMessage, fetchReply: true, components: [row] });
-		// errors handled by internal callbacks
-		awaitDeckValidationButtons(interaction, response, tournament, this.logger, playerDeck);
+		const row = generateDeckValidateButtons(tournament);
+		await interaction.reply({ content: outMessage, components: [row] });
+	}
+}
+
+export function generateDeckValidateButtons(tournament: ManualTournament): ActionRowBuilder<ButtonBuilder> {
+	const row = new ActionRowBuilder<ButtonBuilder>().addComponents(
+		new ButtonBuilder()
+			.setCustomId(encodeCustomId("accept", tournament.tournamentId))
+			.setLabel("Accept")
+			.setStyle(ButtonStyle.Success),
+		new ButtonBuilder()
+			.setCustomId(encodeCustomId("reject", tournament.tournamentId))
+			.setLabel("Reject")
+			.setStyle(ButtonStyle.Danger)
+	);
+	return row;
+}
+
+export class AcceptButtonHandler implements ButtonClickHandler {
+	readonly buttonIds = ["accept"];
+
+	async click(interaction: ButtonInteraction, ...args: string[]): Promise<void> {
+		const tournamentIdString = args[0];
+		const modal = new ModalBuilder()
+			.setCustomId(encodeCustomId("acceptModal", tournamentIdString))
+			.setTitle("Accept Deck");
+		const deckLabelInput = new TextInputBuilder()
+			.setCustomId("acceptDeckLabel")
+			.setLabel("What is the deck's theme?")
+			.setStyle(TextInputStyle.Short)
+			.setRequired(false);
+		const actionRow = new ActionRowBuilder<TextInputBuilder>().addComponents(deckLabelInput);
+		modal.addComponents(actionRow);
+		await interaction.showModal(modal);
+	}
+}
+
+export class RejectButtonHandler implements ButtonClickHandler {
+	readonly buttonIds = ["reject"];
+
+	async click(interaction: ButtonInteraction, ...args: string[]): Promise<void> {
+		const tournamentIdString = args[0];
+		const modal = new ModalBuilder()
+			.setCustomId(encodeCustomId("rejectModal", tournamentIdString))
+			.setTitle("Reject Deck");
+		const rejectReasonInput = new TextInputBuilder()
+			.setCustomId("rejectReason")
+			.setLabel("Why is the deck illegal?")
+			.setStyle(TextInputStyle.Paragraph)
+			.setRequired(false);
+		const actionRow = new ActionRowBuilder<TextInputBuilder>().addComponents(rejectReasonInput);
+		modal.addComponents(actionRow);
+		await interaction.showModal(modal);
+	}
+}
+
+export class AcceptLabelModal implements MessageModalSubmitHandler {
+	readonly modalIds = ["acceptModal"];
+
+	async submit(interaction: ModalMessageModalSubmitInteraction, ...args: string[]): Promise<void> {
+		const tournamentIdString = args[0];
+		const tournamentId = parseInt(tournamentIdString, 10);
+		const deck = await ManualDeckSubmission.findOneOrFail({
+			where: {
+				discordId: interaction.user.id,
+				tournamentId: tournamentId
+			},
+			relations: ["tournament"]
+		});
+		const tournament = deck.tournament;
```

Comment:
don't need this, query ManualTournament on tournamentId _after_ the update query

Reply:
Wait, you're right, if we don't need this, couldn't we just leave out the relation which means we don't run into the bug we need to work around? I just didn't consider the option of making a second query for the tournament because it seemed redundant.